service/history/historyCache.go

```diff
@@ -214,13 +214,16 @@ func (c *historyCache) getOrCreateWorkflowExecutionInternal(
 	//  Consider revisiting this if it causes too much GC activity
 	releaseFunc := c.makeReleaseFunc(key, workflowCtx, forceClearContext)
 
+	lockAcquireTimer := c.metricsClient.StartTimer(scope, metrics.AcqureWorkflowContextLocklatency)
```

Comment:
line 167 - 170 already recorded the time

Reply:
This method also creates context. I was thinking if we should have explicit metric for acquiring lock only.
My understanding that even though similar, CacheLatency might actually record fetching cached value as well so decided to add explicit metric.